tests/src/test/java/alluxio/shell/command/FreeCommandTest.java

```diff
@@ -33,58 +32,64 @@
   @ClassRule
   public static ManuallyScheduleHeartbeat sManuallySchedule =
       new ManuallyScheduleHeartbeat(HeartbeatContext.WORKER_BLOCK_SYNC);
 
   @Test
   public void freeTest() throws IOException, AlluxioException {
-    FileSystemTestUtils.createByteFile(mFileSystem, "/testFile", WriteType.MUST_CACHE, 10);
-    long blockId = mFileSystem.getStatus(new AlluxioURI("/testFile")).getBlockIds().get(0);
+    String fileName = "/testFile";
+    FileSystemTestUtils.createByteFile(mFileSystem, fileName, WriteType.MUST_CACHE, 10);
+    long blockId = mFileSystem.getStatus(new AlluxioURI(fileName)).getBlockIds().get(0);
 
-    mFsShell.run("free", "/testFile");
+    mFsShell.run("free", fileName);
     triggerWorkerHeartbeats(blockId);
-    Assert.assertFalse(
-        mFileSystem.getStatus(new AlluxioURI("/testFile")).getInMemoryPercentage() == 100);
+    Assert.assertFalse(isInMemoryTest(fileName));
   }
 
   @Test
   public void freeWildCardTest() throws IOException, AlluxioException {
     AlluxioShellUtilsTest.resetFileHierarchy(mFileSystem);
-    long blockId =
+    long blockId1 =
         mFileSystem.getStatus(new AlluxioURI("/testWildCards/foo/foobar1")).getBlockIds().get(0);
+    long blockId2 =
+        mFileSystem.getStatus(new AlluxioURI("/testWildCards/foo/foobar2")).getBlockIds().get(0);
 
     int ret = mFsShell.run("free", "/testWild*/foo/*");
 
-    triggerWorkerHeartbeats(blockId);
+    triggerWorkerHeartbeats(blockId1, blockId2);
     Assert.assertEquals(0, ret);
     Assert.assertFalse(isInMemoryTest("/testWildCards/foo/foobar1"));
     Assert.assertFalse(isInMemoryTest("/testWildCards/foo/foobar2"));
     Assert.assertTrue(isInMemoryTest("/testWildCards/bar/foobar3"));
     Assert.assertTrue(isInMemoryTest("/testWildCards/foobar4"));
 
-    blockId =
+    blockId1 =
         mFileSystem.getStatus(new AlluxioURI("/testWildCards/bar/foobar3")).getBlockIds().get(0);
+    blockId2 =
+        mFileSystem.getStatus(new AlluxioURI("/testWildCards/foobar4")).getBlockIds().get(0);
+
     ret = mFsShell.run("free", "/testWild*/*/");
-    triggerWorkerHeartbeats(blockId);
+    triggerWorkerHeartbeats(blockId1, blockId2);
     Assert.assertEquals(0, ret);
     Assert.assertFalse(isInMemoryTest("/testWildCards/bar/foobar3"));
     Assert.assertFalse(isInMemoryTest("/testWildCards/foobar4"));
   }
 
   // Execution of the blocks free needs two heartbeats.
-  private void triggerWorkerHeartbeats(long blockId) {
+  private void triggerWorkerHeartbeats(long... blockIds) {
     try {
       // Schedule 1st heartbeat from worker.
       Assert.assertTrue(HeartbeatScheduler.await(HeartbeatContext.WORKER_BLOCK_SYNC, 5,
           TimeUnit.SECONDS));
       HeartbeatScheduler.schedule(HeartbeatContext.WORKER_BLOCK_SYNC);
 
       // Waiting for the removal of blockMeta from worker.
-      while (mLocalAlluxioCluster.getWorker().getBlockWorker().hasBlockMeta(blockId)) {
-        CommonUtils.sleepMs(50);
+      for (long blockId : blockIds) {
```

Comment:
Done. It's not very clear to me why CommonTestUtils.waitFor is more in favor though, can you talk about that a bit more?